File: frontend/src/api/cleanliness.ts
import { api } from './client'

export interface CleanlinessStats {
  total_audits: number
  start_checks: number
  end_checks: number
  transfers: number
  spills: number
}

export interface CleanlinessAudit {
  id: number
  audit_type: string
  pump_station: string
  operator_name: string
  timestamp: string
  notes: string
  is_spill: boolean
  photos: string[]
}

export const cleanlinessApi = {
  stats: () => api.get<CleanlinessStats>('/cleanliness/stats'),
  audits: () => api.get<CleanlinessAudit[]>('/cleanliness/audits'),
  photoUrl: (filename: string) => `/api/cleanliness/photos/${encodeURIComponent(filename)}`,
  remove: (id: number) => api.del<{ ok: boolean }>(`/cleanliness/audits/${id}`),
}

File: api/schemas/cleanliness.py
from pydantic import BaseModel


class CleanlinessStats(BaseModel):
    total_audits: int
    start_checks: int
    end_checks: int
    transfers: int
    spills: int


class CleanlinessAuditOut(BaseModel):
    id: int
    audit_type: str
    pump_station: str
    operator_name: str
    timestamp: str
    notes: str
    is_spill: bool
    photos: list[str] = []  # filenames; main photo first, then extras
